File: Javascript/Algorithms/1436.js
/**
 * @param {string[][]} paths
 * @return {string}
 */
let destCity = paths => {
    const sources = new Set;
    const dests = new Set;
    for(let path of paths){
        sources.add(path[0]);
        dests.add(path[1]);
    }
    for (let i of dests){
        if (!sources.has(i)){
            return i;
        }
    }
};

File: Rust/Algorithms/1436.rs
use std::collections::HashSet;

impl Solution {
    pub fn dest_city(mut paths: Vec<Vec<String>>) -> String {
        let (mut sources, mut dests) = (HashSet::new(), HashSet::new());
        for mut v in paths {
            // Move elements out of vector, don't need them there.
            let (dest, source) = (v.pop().unwrap(), v.pop().unwrap());
            sources.insert(source);
            dests.insert(dest);
        }
        dests.difference(&sources).nth(0).unwrap().to_owned()
    }
}
